ocaml/xapi/message_forwarding.ml

```diff
@@ -852,13 +852,7 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
            Db.VGPU.set_scheduled_to_be_resident_on ~__context
              ~self:vgpu
              ~value:Ref.null)
-        (Db.VM.get_VGPUs ~__context ~self:vm);
-      List.iter
-        (fun vusb ->
-           Db.VUSB.set_attached ~__context
-             ~self:vusb
-             ~value:Ref.null)
-        (Db.VM.get_VUSBs ~__context ~self:vm)
```

Comment:
I'm not convinced we don't need this - we're setting 'attached' in Vusbops at the same time that we're setting the VGPU scheduled_to_be_resident_on, so presumably we need to clear it on the error path in roughly the same place too?

Reply:
In this function, let forward_to_suitable_host ~local_fn ~__context ~vm ~snapshot ?host_op op =
it will always call  
           if Db.is_valid_ref __context vm
                then clear_scheduled_to_be_resident_on ~__context ~vm))
Then when we start vm which already bind with a vusb, the attached will be cleared . I think this is not as expected.

Reply:
Ah yes, I see - in which case we're probably setting `attached` in the wrong place too. We should move all that logic into VM.start perhaps?

Reply:
I think we can do it in  let allocate_vm_to_host ~__context ~vm ~host ~snapshot ?host_op () =
Originally I add a new function to do this which is ok I think.